### ISSUE TITLE
Error: failed when running ghost commands

### ISSUE THREAD

**Issue by Hsn723** [TryGhost/Ghost-CLI#354]
### This issue is a

* [ X] Bug Report
* [ ] Feature Request

### Summary

After a successful install of ghost via ghost-cli, running ghost commands such as `ghost ls`, `ghost stop` and so on throws an error.

### Steps to Reproduce (for a bug report)

1. Run `ghost ls`, `ghost stop`, `ghost start`, etc.
2. The following is output to console
```
/usr/lib/node_modules/ghost-cli/node_modules/yargs/yargs.js:1079
      else throw err
           ^

Error: failed

    at module.exports.sync (/usr/lib/node_modules/ghost-cli/node_modules/execa/index.js:298:26)
    at handleShell (/usr/lib/node_modules/ghost-cli/node_modules/execa/index.js:100:9)
    at Function.module.exports.shellSync (/usr/lib/node_modules/ghost-cli/node_modules/execa/index.js:307:43)
    at SystemdProcessManager.isRunning (/usr/lib/node_modules/ghost-cli/extensions/systemd/systemd.js:61:19)
    at Instance.get running [as running] (/usr/lib/node_modules/ghost-cli/lib/instance.js:61:27)
    at StartCommand.run (/usr/lib/node_modules/ghost-cli/lib/commands/start.js:27:21)
    at Function._run (/usr/lib/node_modules/ghost-cli/lib/command.js:183:48)
    at Object.handler (/usr/lib/node_modules/ghost-cli/lib/command.js:108:22)
    at Object.self.runCommand (/usr/lib/node_modules/ghost-cli/node_modules/yargs/lib/command.js:233:22)
    at Object.Yargs.self._parseArgs (/usr/lib/node_modules/ghost-cli/node_modules/yargs/yargs.js:990:30)
    at Object.Yargs.self.parse (/usr/lib/node_modules/ghost-cli/node_modules/yargs/yargs.js:532:23)
    at Object.run (/usr/lib/node_modules/ghost-cli/lib/bootstrap.js:153:14)
    at Object.<anonymous> (/usr/lib/node_modules/ghost-cli/bin/ghost:12:11)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```
`ghost version` and `ghost doctor startup` work though.

Additional information which may or may not be relevant:
- Ghost was installed inside a subdirectory of the user's home directory instead of `/var/www`
- Nginx setup was declined and done manually (copied settings from a 0.11.x installation)
- The old 0.11.x blog that I'm planning to migrate is still present, in another directory

### Technical details (will be automatically output by Ghost-CLI if an error occurs):

* OS: Arch Linux Linux version 4.11.9-1-ARCH
* Node Version: v6.11.0
* Ghost-CLI Version: 1.0.0-rc.2
* Environment: production
* Command: `ghost start`

**Comment by acburdine** [comment]
@Hsn723 so it looks like it's failing while checking if `systemd` is running - did you try to set up Ghost with systemd or no?

**Comment by acburdine** [comment]
Note: because you're trying to run it on arch linux rather than the official Ubuntu stack, we may or may not be able to implement a fix. Ghost's systemd configuration is designed to run on Ubuntu 16, so there's no guarantees of it working on arch.

**Comment by Hsn723** [comment]
Ah right, there was also the issue of the systemd service failing, which my sleep-deprived self overlooked. I found the problem, and I don't think it's OS specific. Apparently installing Ghost in `/home/someuser/public_html/blogname` caused a whole bunch of permission issues. I changed the UID specified in the systemd service file to `someuser` and got *some* permission errors to go away, but there was still some errors preventing Ghost to start so I restarted the install from scratch in `/var/www/blogname` and it worked. I can live with not being able to install and run Ghost from my home folder, but that should be specified. Though in some setups I imagine it would be nice to be able to do so.

**Comment by ErisDS** [comment]
cc @sebgie not sure if this is OS specific, but wanted to flag it to you

**Comment by sebgie** [comment]
I have successfully run Ghost from a non-root users home directory on Ubuntu.

@Hsn723 would it be possible to share file/folder permissions for these files/folders:
- `/home`
- `/home/someuser`
- `/home/someuser/public_html`
- `/home/someuser/public_html/blogname`
- `/home/someuser/public_html/blogname/*`

What puzzles me a bit is why `ghost install` would work and `ghost ls` fails afterwards?

**Comment by Hsn723** [comment]
Sure thing:
- `/home`  root:root (drwxr-xr-x)
- `/home/someuser`  someuser:wheel (drwx------)
- `/home/someuser/public_html`  someuser:wheel (drwxr-xr-x)
- `/home/someuser/public_html/blogname`  someuser:wheel (drwxr-xr-x)
- `/home/someuser/public_html/blogname/*`  someuser:wheel (drwxr-xr-x)

Additionally there was some mention about a `/home/ghost/.config` directory in my systemd log but there is no home directory for the user `ghost`. Here's an extract of the `journalctl` log (the same message showed repeatedly so I am copying it only once):
```
┌────────────────────────────────────────

│              ghost-cli update check failed              │
│           Try running with sudo or get access           │
│          to the local update config store via           │
│ sudo chown -R $USER:$(id -gn $USER) /home/ghost/.config │
└────────────────────────────────────────

fs.js:952
  return binding.readdir(pathModule._makeLong(path), options.encoding);
                 ^
Error: EACCES: permission denied, scandir '/home/someuser/public_html/blogname'
    at Error (native)
    at Object.fs.readdirSync (fs.js:952:18)
    at findCandidatesInDir (/usr/lib/node_modules/ghost-cli/node_modules/find-plugins/index.js:66:15)
    at findPlugins (/usr/lib/node_modules/ghost-cli/node_modules/find-plugins/index.js:31:58)
    at findExtensions (/usr/lib/node_modules/ghost-cli/lib/utils/find-extensions.js:26:12)
    at Object.run (/usr/lib/node_modules/ghost-cli/lib/bootstrap.js:69:26)
    at Object.<anonymous> (/usr/lib/node_modules/ghost-cli/bin/ghost:12:11)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3
```
Further, the permissions for `/home/someuser/.config` is someuser:wheel (drwxr-xr-x) and the `/etc/passwd` entry for `ghost` is `ghost:x:994:994::/home/ghost:/bin/bash`.

**Comment by sebgie** [comment]
Thank you for the information!

I'm going to X-reference this to #355. Ubuntu has different permissions for `/home/someuser`:

```
drwxr-xr-x  3 root      root      4096 Jul 13 06:27 ./
drwxr-xr-x 23 root      root      4096 Jul 13 11:31 ../
drwxr-xr-x  6 someuser someuser 4096 Jul 13 11:26 someuser/
```

@acburdine any idea where the `/home/ghost/.config` file would come from?

**Comment by acburdine** [comment]
yeah - that comes from update-notifier. I'll add a fix to not run the update check during `ghost run`.

**Comment by ErisDS** [comment]
Just for clarity here, afaik there is no `/home/ghost/.config` file

Ghost CLI creates files inside a folder located at `~/.ghost/` including `~/.ghost/config` and `~/.ghost/logs/*`. ~ is the home folder of the user who runs the `ghost x` commands.

Would it perhaps be useful to give these files more permissions than we do to ghost itself? E.g. 775 rather than 755?

**Comment by acburdine** [comment]
There is a `/home/ghost/.config` file, created by the `update-notifier` package that ghost-cli uses, which is why there's the "ghost-cli update check failed" error above.

FWIW the two problems mentioned here are addressed in both #356 and #277

**Comment by ErisDS** [comment]
> There is a /home/ghost/.config file

OK, I consider this a separate issue. Having `~/.ghost/config` and `/home/ghost/.config` is WAAAAY too confusing. I assumed it was a typo.
